chain-signatures/node/src/http_client.rs

```diff
@@ -162,54 +166,58 @@ impl MessageQueue {
             encrypted.push((encrypted_msg, (info, msg, instant)));
         }
 
-        let mut compacted = 0;
+        let mut tasks = tokio::task::JoinSet::new();
         for (id, encrypted) in encrypted {
             for partition in partition_ciphered_256kb(encrypted) {
-                let (encrypted_partition, msgs): (Vec<_>, Vec<_>) = partition.into_iter().unzip();
+                let (encrypted_partition, _msgs): (Vec<_>, Vec<_>) = partition.into_iter().unzip();
                 // guaranteed to unwrap due to our previous loop check:
-                let info = participants.get(&Participant::from(id)).unwrap();
+                let id = Participant::from(id);
+                let info = participants.get(&id).unwrap();
                 let account_id = &info.account_id;
 
-                let start = Instant::now();
                 crate::metrics::NUM_SEND_ENCRYPTED_TOTAL
                     .with_label_values(&[account_id.as_str()])
                     .inc();
-                if let Err(err) = send_encrypted(
+
+                tasks.spawn(send_encrypted(
                     from,
-                    client,
-                    &info.url,
+                    client.clone(),
+                    info.url.clone(),
                     encrypted_partition,
                     Duration::from_millis(self.message_options.timeout),
-                )
-                .await
-                {
-                    crate::metrics::NUM_SEND_ENCRYPTED_FAILURE
-                        .with_label_values(&[account_id.as_str()])
-                        .inc();
-                    crate::metrics::FAILED_SEND_ENCRYPTED_LATENCY
-                        .with_label_values(&[account_id.as_str()])
-                        .observe(start.elapsed().as_millis() as f64);
-
-                    // since we failed, put back all the messages related to this
-                    failed.extend(msgs);
+                ));
+            }
+        }
+
+        let mut compacted = 0;
+        while let Some(result) = tasks.join_next().await {
+            match result {
+                Ok(result) => {
+                    let Err(err) = result else {
+                        compacted += 1;
+                        continue;
+                    };
                     errors.push(err);
-                } else {
-                    compacted += msgs.len();
-                    crate::metrics::SEND_ENCRYPTED_LATENCY
-                        .with_label_values(&[account_id.as_str()])
-                        .observe(start.elapsed().as_millis() as f64);
+                }
+                Err(err) => {
+                    tracing::error!(?err, "message queue task failure");
                 }
             }
         }
 
-        if uncompacted > 0 {
+        let elapsed = outer.elapsed();
+        if elapsed > Duration::from_millis(100) && uncompacted > 0 {
```

Comment:
I guess this means if less than 100 ms has passed then sent messages won't be logged? I'd prefer to still see how many messages have been sent if any, even if the time taken is very fast.

Reply:
whoops, that was not supposed to make it to the PR. This was just me debugging locally